score-xml/src/main/java/io/cloudslang/content/xml/utils/XmlUtils.java

```diff
@@ -109,4 +265,93 @@ private static String transformElementNode(Node node) throws TransformerExceptio
+        }
+        return doc;
+    }
+
+    public static NamespaceContext getNamespaceContext(CommonInputs commonInputs, Document doc) throws Exception {
+        NamespaceContext context;
+        if (Constants.XML_PATH.equalsIgnoreCase(commonInputs.getXmlDocumentSource())) {
+            StringWriter writer = XmlUtils.getStringWriter(doc);
+            context = XmlUtils.getNamespaceContext(writer.toString(), "");
+        } else {
+            context = XmlUtils.getNamespaceContext(commonInputs.getXmlDocument(), "");
+        }
+        return context;
+    }
+
+    public static String createXmlDocumentFromUrl(CommonInputs commonInputs) throws ParserConfigurationException, SAXException, IOException {
+        if (commonInputs.getXmlDocument().startsWith("http")) {
+            ScoreHttpClient scoreHttpClient = new ScoreHttpClient();
+            HttpClientInputs httpClientInputs = new HttpClientInputs();
+            httpClientInputs.setMethod(HttpGet.METHOD_NAME);
+            httpClientInputs.setUrl(commonInputs.getXmlDocument());
+
+            if (commonInputs.getUsername().isEmpty()) {
+                httpClientInputs.setAuthType(AuthTypes.ANONYMOUS);
+            } else {
+                httpClientInputs.setAuthType(AuthTypes.BASIC);
+                httpClientInputs.setUsername(commonInputs.getUsername());
+                httpClientInputs.setPassword(commonInputs.getPassword());
+            }
+            httpClientInputs.setRequestCharacterSet(StandardCharsets.UTF_8.toString());
+            httpClientInputs.setResponseCharacterSet(StandardCharsets.UTF_8.toString());
+            httpClientInputs.setTrustAllRoots(commonInputs.getTrustAllRoots());
+            httpClientInputs.setKeystore(commonInputs.getKeystore());
+            httpClientInputs.setKeystorePassword(commonInputs.getKeystorePassword());
+            httpClientInputs.setTrustKeystore(commonInputs.getTrustKeystore());
+            httpClientInputs.setTrustPassword(commonInputs.getTrustPassword());
+            httpClientInputs.setX509HostnameVerifier(commonInputs.getX509Hostnameverifier());
+            httpClientInputs.setProxyHost(commonInputs.getProxyHost());
+            httpClientInputs.setProxyPort(commonInputs.getProxyPort());
+            httpClientInputs.setProxyUsername(commonInputs.getProxyUsername());
+            httpClientInputs.setProxyPassword(commonInputs.getProxyPassword());
+
+            Map<String, String> requestResponse = scoreHttpClient.execute(httpClientInputs);
+            if (!OK_STATUS_CODE.equals(requestResponse.get(ScoreHttpClient.STATUS_CODE))) {
+                throw new RuntimeException("Http request to specified URL: " + commonInputs.getXmlDocument() + " failed with status code: " + requestResponse.get(ScoreHttpClient.STATUS_CODE) + ". Request response is: " + requestResponse.get(Constants.Outputs.RETURN_RESULT));
+            }
+            return requestResponse.get(Constants.Outputs.RETURN_RESULT);
+        } else {
```

Comment:
Why is this code(that creates a Document from file) here? There is a specific method for that: XmlUtils.createDocumentFromFile().